### PR TITLE
update RobotName to never version from JS repo

### PR DESCRIPTION
**Pull request by joshgoebel**

### CODE REVIEW

**Comment by yyyc514** [comment]
No idea on this bot error.

File: exercises/robot-name/robot-name.example.ts [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+      throw "no more names"
+
+    const randomPosition = rand(this.availableNames.length)
+    const name = this.availableNames[randomPosition]
+    const lastName = this.availableNames.pop()!
+    // swap the last name into the position of the name
+    // we just removed (unless we happened to randomly
+    // pick the very last name already)
+    if (name !== lastName)
+      this.availableNames[randomPosition] = lastName
```
Comment (yyyc514) [inline exercises/robot-name/robot-name.example.ts:44]: Is this better?

```ts
    // if we have selected any other name than the last
    // name then place the last name in the blank spot
    // left by the name we just "removed"
    // (this is a fast adoption of splice)
    const lastIndex = this.availableNames.length - 1
    const lastName = this.availableNames.pop()!
    if (randomPosition !== lastIndex) {
      this.availableNames[randomPosition] = lastName
    }
```

**Comment by yyyc514** [comment]
@SleeplessByte Any idea on how I move forward here?

**Comment by SleeplessByte** [comment]
@joshgoebel fixed it!

**Review by SleeplessByte — APPROVED** [review]

**Pull request merged**